### PR TITLE
Get rid of `location.siblings` call

### PR DESCRIPTION
**Pull request by esoergel**
@proteusvacuum

### CODE REVIEW

**Comment by esoergel** [comment]
Oh whoops, @benrudolph is my code buddy

File: corehq/apps/locations/forms.py [hunk 1]
```diff
@@ -195,10 +195,14 @@ def clean_parent_id(self):
...
         name = self.cleaned_data['name']
+        parent = self.cleaned_data.get('parent')
 
         if self.strict:
-            siblings = self.location.siblings(self.cleaned_data.get('parent'))
-            if name in [loc.name for loc in siblings]:
+            if (SQLLocation.objects.filter(domain=self.location.domain,
+                                           parent=parent,
+                                           name=name)
+                                   .exclude(location_id=self.location._id)
```
Comment (benrudolph) [inline corehq/apps/locations/forms.py]: does this mean we have locations that reference themselves as parents? haha

**Comment by benrudolph** [comment]
👍 on ✔️

File: corehq/apps/locations/forms.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/locations/forms.py]: I certainly hope not!
This is getting all locations which list `parent` as the `parent`, but excludes the current location.  The idea is, when setting the parent, make sure that there are no other children of that parent with the same name.

**Comment by snopoke** [comment]
nice to see some locations cleanup happening!

**Comment by proteusvacuum** [comment]
kicked build

**Comment by benrudolph** [comment]
```
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/ilsgateway/tests/test_report_reminders.py", line 136, in test_basic_report_no_responses
    result = construct_randr_summary(self.district)
  File "/mnt/commcare-hq-ro/custom/ilsgateway/tanzania/reminders/reports.py", line 61, in construct_randr_summary
    get_business_day_of_month_before(now.year, now.month, 5)
  File "/mnt/commcare-hq-ro/custom/ilsgateway/tanzania/reminders/reports.py", line 50, in construct_summary
    ret["total"] = len(locations)
TypeError: object of type 'itertools.imap' has no len()
```
could be related to removing `list` from the output of `get_children` though didn't look very hard

**Comment by esoergel** [comment]
:white_check_mark:

**Comment by snopoke** [comment]
🎈 Nice

**Pull request merged**